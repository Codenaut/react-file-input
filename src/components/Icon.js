import React from 'react';
import PropTypes from 'prop-types';

const Icon = ({ icon }) => {
  const [width, height, , , path] = icon;

  return (
    <svg
      style={{
        display: 'inline-block',
        fontSize: 'inherit',
        height: '1em',
        overflow: 'visible',
        verticalAlign: '-.125em',
        width: '1em',
      }}
      viewBox={`0 0 ${width} ${height}`}
    >
      <path d={path} fill="currentColor"/>
    </svg>
  );
};

Icon.propTypes = {
  icon: PropTypes.arrayOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.array,
  ])).isRequired,

};

export default Icon;
