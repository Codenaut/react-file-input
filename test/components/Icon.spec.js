import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Icon from '../../src/components/Icon';

const defaultProps = {
  icon: [
    128,
    256,
    [],
    'string',
    'path',
  ],
};

const setup = (props = {}) => {
  const icon = shallow(<Icon {...defaultProps} {...props} />);

  return {
    icon,
    svg: icon.find('svg'),
    path: icon.find('path'),
  };
};

describe('components', () => {
  describe('Icon', () => {
    it('should render a svg', () => {
      const { svg } = setup();

      expect(svg).toHaveLength(1);
      expect(svg.prop('viewBox')).toBe(`0 0 ${defaultProps.icon[0]} ${defaultProps.icon[1]}`);
    });

    it('should render a path', () => {
      const { path } = setup();

      expect(path).toHaveLength(1);
      expect(path.prop('d')).toBe(defaultProps.icon[4]);
      expect(path.prop('fill')).toBe('currentColor');
    });

    it('should match exact snapshot', () => {
      const tree = renderer.create(
        <div>
          <Icon {...defaultProps} />
        </div>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
