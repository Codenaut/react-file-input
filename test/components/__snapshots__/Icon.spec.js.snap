// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Icon should match exact snapshot 1`] = `
<div>
  <svg
    style={
      Object {
        "display": "inline-block",
        "fontSize": "inherit",
        "height": "1em",
        "overflow": "visible",
        "verticalAlign": "-.125em",
        "width": "1em",
      }
    }
    viewBox="0 0 128 256"
  >
    <path
      d="path"
      fill="currentColor"
    />
  </svg>
</div>
`;
